### PR TITLE
DEVDOCS-3854 graphql faceted textual

### PR DESCRIPTION
**Pull request by bc-andreadao**
# [DEVDOCS-3854](https://bigcommercecloud.atlassian.net/browse/DEVDOCS-3854)
-New doc for Storefront GraphQL Faceted and Textual Search
-Merge when goes GA

Story Ticket: [DEVDOCS-3760](https://bigcommercecloud.atlassian.net/browse/DEVDOCS-3760)

### CODE REVIEW

**Review by bc-dima-pasieka — APPROVED** [review]
👍

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 1]
```diff
@@ -0,0 +1,982 @@
...
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
```
Comment (slsriehl) [inline docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:88]:
```suggestion
```
&nbsp;
```

Comment (slsriehl) [reply docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:88]: disregard that backslash --

the comment is that these blocks will nest without a `&nbsp;` between them.

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 2]
```diff
@@ -0,0 +1,982 @@
...
```
Comment (slsriehl) [inline docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:157]: and here, `&nbsp;` if nesting not desired.

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 3]
```diff
@@ -0,0 +1,982 @@
...
+               }
+             }
+           }
+         }
+      }
+    }
+  }
+}
+```
+
```
Comment (slsriehl) [inline docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:355]: and here, `&nbsp;` if nesting not desired.

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 4]
```diff
@@ -0,0 +1,982 @@
...
```
Comment (slsriehl) [inline docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:699]: and here, `&nbsp;` if nesting not desired.

you the picture.

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md (hunk 1 above)
Comment (bc-andreadao) [reply docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:93]: Used new feature to nest while choosing the tab name

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md (hunk 2 above)
Comment (bc-andreadao) [reply docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:174]: Used new feature to nest while choosing the tab name

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md (hunk 4 above)
Comment (bc-andreadao) [reply docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:744]: Used new feature to nest while choosing the tab name

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 5]
```diff
@@ -0,0 +1,612 @@
...
+<!-- type: tab-end -->
+
+The `sort` affects only the list of products returned. A merchant's [Product Filtering settings](https://support.bigcommerce.com/s/article/Product-Filtering-Settings?language=en_US#setup) determines how facets are sorted. 
+
+<!-- theme: info -->
+> **Note**
```
Comment (slsriehl) [inline docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:224]:
```suggestion
> #### Note
```

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 6]
```diff
@@ -0,0 +1,612 @@
+# Storefront GraphQL's Faceted and Textual Search
+
+BigCommerce storefront search features are available in the Storefront GraphQL API so that storefront API consumers can use BigCommerce's backend search to rebuild their faceted and textual search UI. These built-in capabilities allow Stencil developers to build on top of our search engine, as well as merchants on headless storefronts to access search, sort, and filtering capabilities available from BigCommerce. 
+
+With Storefront GraphQL's Faceted and Textual Search, you can:
+- Load category pages with no selections, including the both the facets and products relevant to the search results 
+- Load category pages with facet selections for specific facets
+- Load featured products from specific categories
+- Quicksearch products by using search terms (textual search)
```
Comment (slsriehl) [inline docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:9]: Is `quicksearch` a technical term? @bc-andreadao

Comment (bc-andreadao) [reply docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:9]: I didn't mean it in a technical way, maybe quickly search would be better

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 7]
```diff
@@ -0,0 +1,612 @@
+# Storefront GraphQL's Faceted and Textual Search
+
+BigCommerce storefront search features are available in the Storefront GraphQL API so that storefront API consumers can use BigCommerce's backend search to rebuild their faceted and textual search UI. These built-in capabilities allow Stencil developers to build on top of our search engine, as well as merchants on headless storefronts to access search, sort, and filtering capabilities available from BigCommerce. 
+
+With Storefront GraphQL's Faceted and Textual Search, you can:
+- Load category pages with no selections, including the both the facets and products relevant to the search results
```
Comment (markcmurphy) [inline docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:6]:
```suggestion
- Load category pages with no selections, including both the facets and products relevant to the search results 
```

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 8]
```diff
@@ -0,0 +1,612 @@
+# Storefront GraphQL's Faceted and Textual Search
+
+BigCommerce storefront search features are available in the Storefront GraphQL API so that storefront API consumers can use BigCommerce's backend search to rebuild their faceted and textual search UI. These built-in capabilities allow Stencil developers to build on top of our search engine, as well as merchants on headless storefronts to access search, sort, and filtering capabilities available from BigCommerce. 
+
+With Storefront GraphQL's Faceted and Textual Search, you can:
+- Load category pages with no selections, including the both the facets and products relevant to the search results 
+- Load category pages with facet selections for specific facets
+- Load featured products from specific categories
+- Quickly search products by using search terms (textual search)
+- Sort products alphabetically, by newest to oldest, etc
```
Comment (markcmurphy) [inline docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:10]:
```suggestion
- Sort products alphabetically, from newest to oldest, and more
```

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 9]
```diff
@@ -0,0 +1,612 @@
...
+    ],
+    isFreeShipping:true,
+    isFeatured:true,
+    isInStock:true
+  }
+  ... 
+) { 
+...
+```
+These filters affect **both** the products and facets that are returned. For example, filtering by rating returns only products that are within the specified rating range, as well as only facets that have products within the rating range.
```
Comment (markcmurphy) [inline docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:51]:
```suggestion
These filters affect **both** the products and facets that are returned. For example, filtering by rating returns only products within the specified rating range and only facets that have products within the rating range.
```
Suggestion to make this a bit more concise

**Review by markcmurphy — COMMENTED** [review]
I am still working through the tutorial, but I am submitting some comments I have pending now

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 10]
```diff
@@ -0,0 +1,612 @@
...
+  }
+  ... 
+) { 
+...
+```
+These filters affect **both** the products and facets that are returned. For example, filtering by rating returns only products that are within the specified rating range, as well as only facets that have products within the rating range. 
+
+See [GraphQL Playground](/graphql-playground) for descriptions of each filter. 
+
+## How to Get Products
```
[the remark this answers is not in the document]
Comment (bc-andreadao) [reply docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:55]: I have changed it in a PR that will merge into this branch (https://github.com/bigcommerce/dev-docs/pull/1512)

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 11]
```diff
@@ -0,0 +1,612 @@
...
+        }
+      }
+    }
+  }
+}
+```
+
+<!-- type: tab-end -->
+
+## Related Resources
```
[the remark this answers is not in the document]
Comment (bc-andreadao) [reply docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:609]: I have changed the heading in a PR that will merge into this branch (https://github.com/bigcommerce/dev-docs/pull/1512)

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 12]
```diff
@@ -0,0 +1,612 @@
+# Storefront GraphQL's Faceted and Textual Search
+
+BigCommerce storefront search features are available in the Storefront GraphQL API so that storefront API consumers can use BigCommerce's backend search to rebuild their faceted and textual search UI. These built-in capabilities allow Stencil developers to build on top of our search engine, as well as merchants on headless storefronts to access search, sort, and filtering capabilities available from BigCommerce.
```
[the remark this answers is not in the document]
Comment (bc-andreadao) [reply docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md]: I have added a definition for faceted and textual search

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 13]
```diff
@@ -0,0 +1,612 @@
+# Storefront GraphQL's Faceted and Textual Search
+
+BigCommerce storefront search features are available in the Storefront GraphQL API so that storefront API consumers can use BigCommerce's backend search to rebuild their faceted and textual search UI. These built-in capabilities allow Stencil developers to build on top of our search engine, as well as merchants on headless storefronts to access search, sort, and filtering capabilities available from BigCommerce. 
+
+With Storefront GraphQL's Faceted and Textual Search, you can:
```
[the remark this answers is not in the document]
Comment (bc-andreadao) [reply docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md]: I have updated the line to a introduce the list with a complete sentence.

**Comment by bc-andreadao** [comment]
> I am still working through the tutorial, but I am submitting some comments I have pending now

Thanks, I made your suggestions. I re-requested your review so it would show up on your list of PRs

File: toc.json [hunk 14]
```diff
@@ -78,24 +78,30 @@
...
           "flag": "code samples",
           "uri": "docs/api-docs/storefront/graphql/graphql-api-examples.md"
         },
+        // {
+        //  "type": "item",
+        //  "title": "Storefront GraphQL Explorer",
+        //  "href": "",
+        //  "uri": ""
+        // },
         {
```
Comment (slsriehl) [inline toc.json:100]:
```suggestion
        {
          "type": "item",
          "title": "GraphQL Storefront Explorer",
          "href": "/api-docs/storefront/graphql/explorer",
          "uri": "https://developer.bigcommerce.com/graphql-explorer"
        },
        {

```

explorer is back up!

File: toc.json [hunk 15]
```diff
@@ -78,24 +78,30 @@
...
           "type": "item",
-          "title": "GraphQL Playground",
+          "title": "Storefront GraphQL Playground",
```
Comment (slsriehl) [inline toc.json:102]:
```suggestion
          "title": "GraphQL Storefront Playground",
```

Comment (slsriehl) [reply toc.json:102]: per convention that Mark & I set during heavy revisions a couple weeks ago.

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 16]
```diff
@@ -0,0 +1,612 @@
...
+  }
+}
+```
+<!-- type: tab-end -->
+
+For a complete list of facets that can be returned, see the [GraphQL Storefront Playground](/graphql-playground).
+
+## Put it all together: get products and facets
+
+To get both products and facets, specify `products` and `filters` as a field in `searchProducts`.
```
Comment (slsriehl) [inline docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:375]:
```suggestion
To get both products and facets, specify both `products` and `filters` fields in `searchProducts`. 
```
subject-verb agreement

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 17]
```diff
@@ -0,0 +1,612 @@
...
+          }
+        }
+      }
+    }
+  }
+}
+```
+<!-- type: tab-end -->
+
+For a complete list of facets that can be returned, see the [GraphQL Storefront Playground](/graphql-playground).
```
Comment (slsriehl) [inline docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:371]:
```suggestion
For a complete list of facets that can be returned, see the [GraphQL Storefront Playground](https://developer.bigcommerce.com/graphql-playground).
```
the playground isn't part of the Stoplight docs app, so it needs the whole URL. (this is also makes it open in a new window, which is great IMO for its use case.)

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 18]
```diff
@@ -0,0 +1,612 @@
...
+  }
+}
+```
+
+<!-- type: tab-end -->
+
+## Resources
+
+- [GraphQL Storefront API Overview](/api-docs/storefront/graphql/graphql-storefront-api-overview)
+- [GraphQL Storefront Playground](/graphql-playground)
```
Comment (slsriehl) [inline docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:612]:
```suggestion
- [GraphQL Storefront Playground](https://developer.bigcommerce.com/graphql-playground)
```
the playground isn't part of the Stoplight docs app, so it needs the whole URL. (this is also makes it open in a new window, which is great IMO for its use case.)

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 19]
```diff
@@ -0,0 +1,612 @@
...
+}
+```
+
+<!-- type: tab-end -->
+
+The `sort` affects only the list of products returned. A merchant's [Product Filtering settings](https://support.bigcommerce.com/s/article/Product-Filtering-Settings?language=en_US#setup) determines how facets are sorted. 
+
+<!-- theme: info -->
+> #### Note
+> For a list of product fields that `searchTerm` searches, see [Store Search Product Fields](https://support.bigcommerce.com/s/article/Store-Search?language=en_US#best-practices).
```
Comment (slsriehl) [inline docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:225]:
```suggestion
> For a list of product fields that `searchTerm` searches, see the [Store Search Product Fields](https://support.bigcommerce.com/s/article/Store-Search?language=en_US#best-practices) article.
```

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 20]
```diff
@@ -0,0 +1,612 @@
...
+        }
+      }
+    }
+  }
+}
+```
+
+<!-- type: tab-end -->
+
+The `sort` affects only the list of products returned. A merchant's [Product Filtering settings](https://support.bigcommerce.com/s/article/Product-Filtering-Settings?language=en_US#setup) determines how facets are sorted.
```
Comment (slsriehl) [inline docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:221]:
```suggestion
The `sort` affects only the list of products returned. A merchant's [Product Filtering settings](https://support.bigcommerce.com/s/article/Product-Filtering-Settings?language=en_US#setup) determine how facets are sorted. 
```
subject-verb agreement

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 21]
```diff
@@ -0,0 +1,612 @@
...
+          }
+        }
+      }
+    }
+  }
+}
+```
+<!-- type: tab-end -->
+
+You can sort the products that are returned using the `sort` field. Here is an example request that searches for products using a search term and sorts the products in alphabetical order:
```
Comment (slsriehl) [inline docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:140]:
```suggestion
You can sort the products that are returned using the `sort` field. Here is an example query that searches for products using a search term and sorts the returned products in alphabetical order:
```
I had to do some mental work to parse this sentence after `that searches`.

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 22]
```diff
@@ -0,0 +1,612 @@
...
+
+The GraphQL Storefront API's faceted and textual search lets you create the following features:
+- Load category pages with no selections, including both the facets and products relevant to the search results 
+- Load category pages with facet selections for specific facets
+- Load featured products from specific categories
+- Quickly search products by using search terms (textual search)
+- Sort products alphabetically, from newest to oldest, and more
+
+You can access these features by querying the `SearchProducts` field. Note that you can query facets, filter by rating, or filter by "in-stock" only if the merchant is on a Pro or Enterprise plan. A merchant must [enable Product Filtering](https://support.bigcommerce.com/s/article/Product-Filtering-Settings?language=en_US#setup) for facets to be returned. In addition, only facets that a merchant marks as visible in their Product Filtering settings will be returned.
```
Comment (slsriehl) [inline docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:12]:
```suggestion
You can access these features by querying the `SearchProducts` field. Note that you can query facets, filter by rating, or filter by "in-stock" only if the merchant is on a Pro or Enterprise plan. A merchant must [Enable Product Filtering](https://support.bigcommerce.com/s/article/Product-Filtering-Settings?language=en_US#setup) for facets to be returned. In addition, only facets that a merchant marks as visible in their Product Filtering settings will be returned. 
```
I think that this article title cases the other references to KB articles.

File: docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md [hunk 23]
```diff
@@ -0,0 +1,612 @@
+# Faceted and Textual Search with the GraphQL Storefront API
+
+Faceted search is a method of helping shoppers navigate a store's product inventory by categorizing products into various categories, brands, product features, and more. Textual search is a method of retrieving products based on product fields, for example, product name and description. BigCommerce's GraphQL Storefront API lets you power your storefront's faceted and textual search with results from our back-end search engine. These built-in capabilities allow Stencil developers to build on top of our search engine, and this API enables merchants on headless storefronts to use BigCommerce's search, sort, and filtering capabilities.
```
Comment (slsriehl) [inline docs/api-docs/storefront/graphql/graphql-faceted-textual-search.md:3]:
```suggestion
Faceted search is a method of helping shoppers navigate a store's product inventory by categorizing products into various categories, brands, product features, and more. Textual search is a method of retrieving products based on product fields, for example, product name and description. BigCommerce's GraphQL Storefront API lets merchants on headless storefronts use faceted and textual search powered by results from our back-end search engine. These built-in capabilities also allow Stencil developers to build on top of our search engine. 
```
it was unclear which API

**Review by slsriehl — APPROVED** [review]
most excellent.

**Pull request merged**